### PR TITLE
[10.x] PHP 8.1 Support - Refactored Content Handling with `match` Expressions in `Illuminate\Http\Response`

### PR DESCRIPTION
**Pull request by miladev95**
This pull request introduces PHP 8.1 support by refactoring content handling methods within the codebase. The changes utilize the new `match` expression, improving readability and leveraging PHP 8.1's capabilities for better maintenance and future compatibility.

Changes:
- Updated `shouldBeJson` method to use `match` for clarity and brevity.
- Restructured `morphToJson` method to use `match` for better code organization and readability.

This enhancement provides a more concise and maintainable approach for content handling, ensuring compatibility with the PHP 8.1 version while maintaining Laravel's high coding standards.

### CODE REVIEW

File: src/Illuminate/Http/Response.php [hunk 1]
```diff
@@ -82,11 +82,14 @@ public function setContent(mixed $content): static
      */
     protected function shouldBeJson($content)
     {
-        return $content instanceof Arrayable ||
-               $content instanceof Jsonable ||
-               $content instanceof ArrayObject ||
-               $content instanceof JsonSerializable ||
-               is_array($content);
+        return match (true) {
```
Comment (PerryvanderMeer) [inline src/Illuminate/Http/Response.php:85]: Not sure if this is a real improvement for the readability..

**Comment by milwad-dev** [comment]
👎 I don't agree
Is not refactoring!!

**Comment by miladev95** [comment]
> 👎 I don't agree Is not refactoring!!

why? 🤔

File: src/Illuminate/Http/Response.php (hunk 1 above)
Comment (miladev95) [reply src/Illuminate/Http/Response.php:85]: The match expression provides a clearer delineation between the different conditions and their outcomes, which could make it easier for developers to understand the logic at a glance. The goal was to simplify and streamline the conditional checks for better maintenance and understanding in the long run.

Comment (crynobone) [reply src/Illuminate/Http/Response.php:85]: I would argue it harder to read now.

File: src/Illuminate/Http/Response.php [hunk 2]
```diff
@@ -97,12 +100,10 @@ protected function shouldBeJson($content)
...
-        } elseif ($content instanceof Arrayable) {
-            return json_encode($content->toArray());
-        }
-
-        return json_encode($content);
+        return match (true) {
+            $content instanceof Jsonable => $content->toJson(),
+            $content instanceof Arrayable => json_encode($content->toArray()),
+            default => json_encode($content),
+        };
```
Comment (mpyw) [inline src/Illuminate/Http/Response.php:107]: I like this

File: src/Illuminate/Http/Response.php [hunk 3]
```diff
@@ -82,11 +82,14 @@ public function setContent(mixed $content): static
...
-               $content instanceof JsonSerializable ||
-               is_array($content);
+        return match (true) {
+            $content instanceof Arrayable,
+            $content instanceof Jsonable,
+            $content instanceof ArrayObject,
+            $content instanceof JsonSerializable,
+            is_array($content) => true,
+            default => false,
+        };
```
Comment (mpyw) [inline src/Illuminate/Http/Response.php:92]: But I hate this

Comment (mpyw) [reply src/Illuminate/Http/Response.php:92]: You're probably writing the one equivalent to:

```php
if (
  $content instanceof Arrayable
  || $content instanceof Jsonable
  || $content instanceof ArrayObject
  || $content instanceof JsonSerializable
  || is_array($content)
) {
    return true;
} else {
    return false;
}
```

**Pull request closed without merging**